data/scripting/win_conditions/win_condition_functions.lua

```diff
@@ -336,40 +336,79 @@ function rank_players(all_player_points, plrs)
-      time = (ngettext("%i minute", "%i minutes", h, m)):bformat(m)
-      -- TRANSLATORS: Context: 'The game will end in 2 hours and 30 minutes.'
-      time = (ngettext("%1% hour and %2%", "%1% hours and %2%", h, m)):bformat(h, time)
-   elseif m > 0 then
-      -- TRANSLATORS: Context: 'The game will end in 30 minutes.'
-      time = (ngettext("%i minute", "%i minutes", m)):bformat(m)
+   local d = math.floor(remaining_time / (60 * 24))
+   remaining_time = remaining_time - (d * 60 * 24)
+   local h = math.floor(remaining_time / 60)
+   remaining_time = remaining_time - (h * 60)
+
+   if d > 0 then
+      if ((h > 0) and (remaining_time > 0)) then
+         -- TRANSLATORS: Context: 'The game will end in 2 hours and 30 minutes.'
+         local time_m = (ngettext("%i minute", "%i minutes", remaining_time)):bformat(remaining_time)
+         -- TRANSLATORS: Context: 'The game will end in 2 hours and 30 minutes.'
+         time = (ngettext("%1% hour", "%1% hours", h)):bformat(h)
+         -- TRANSLATORS: Context: 'The game will end in 2 days, 1 hour, and 30 minutes.'
+         time = (ngettext("%1% day, %2%, and %3%", "%1% days, %2%, and %3%", d)):bformat(d, time, time_m)
+      elseif h > 0 then
+         -- TRANSLATORS: Context: 'The game will end in 2 days and 3 hours.'
+         time = (ngettext("%i hour", "%i hours", h)):bformat(h)
+         -- TRANSLATORS: Context: 'The game will end in 2 days and 3 hours.'
+         time = (ngettext("%1% day and %2%", "%1% days and %2%", d)):bformat(d, time)
+      elseif remaining_time > 0 then
+         -- TRANSLATORS: Context: 'The game will end in 2 days and 30 minutes.'
+         time = (ngettext("%i minute", "%i minutes", remaining_time)):bformat(remaining_time)
+         -- TRANSLATORS: Context: 'The game will end in 2 days and 30 minutes.'
+         time = (ngettext("%1% day and %2%", "%1% days and %2%", d)):bformat(d, time)
+      else
+         -- TRANSLATORS: Context: 'The game will end in 2 days.'
+         time = (ngettext("%1% day", "%1% days", d)):bformat(d)
+      end
    else
-      -- TRANSLATORS: Context: 'The game will end in 2 hours.'
-      time = (ngettext("%1% hour", "%1% hours", h)):bformat(h)
+      if ((h > 0) and (remaining_time > 0)) then
+         -- TRANSLATORS: Context: 'The game will end in 2 hours and 30 minutes.'
+         time = (ngettext("%i minute", "%i minutes", remaining_time)):bformat(remaining_time)
+         -- TRANSLATORS: Context: 'The game will end in 2 hours and 30 minutes.'
+         time = (ngettext("%1% hour and %2%", "%1% hours and %2%", h)):bformat(h, time)
+      elseif remaining_time > 0 then
+         -- TRANSLATORS: Context: 'The game will end in 30 minutes.'
+         time = (ngettext("%i minute", "%i minutes", remaining_time)):bformat(remaining_time)
+      else
+         -- TRANSLATORS: Context: 'The game will end in 2 hours.'
+         time = (ngettext("%1% hour", "%1% hours", h)):bformat(h)
+      end
```

Comment:
Can't we just create the strings for `d`, `h` and `remaining_time` (ie. minutes) once first unconditionally, and then only combine those as needed?